### PR TITLE
Add the CLI for pulp_ostree

### PR DESCRIPTION
**Pull request by lubosmj**
[noissue]

### CODE REVIEW

File: pulpcore/cli/ostree/distribution.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+            # unset repository or repository version
+            if distribution["repository"]:
+                body["repository"] = ""
+            elif distribution["repository_version"]:
+                body["repository_version"] = ""
+        else:
+            repository = cast(PulpEntityContext, repository)
+            if version is not None:
+                if distribution["repository"]:
+                    distribution_ctx.update(href, body={"repository": ""}, non_blocking=True)
```
[the remark this answers is not in the document]
Comment (mdellweg) [reply pulpcore/cli/ostree/distribution.py:103]: Ah, the problem, this code tries to work around is, when you switch from a repository to a repository-version and back.

[the remark this answers is not in the document]
Comment (lubosmj) [reply pulpcore/cli/ostree/distribution.py]: In the `test (3.6, 0.12.0, latest, latest, Containerfile)` runner, it fails with the following error:
```
Error: {"non_field_errors":["Only one of the attributes 'repository' and 'repository_version' may be used simultaneously."]}
```

However, the `test (3.9, nightly, latest, Containerfile.source)` runner always succeeds (see https://github.com/pulp/pulp-cli-ostree/actions/runs/1420250606).

Do you think I should use two update calls once again?

[the remark this answers is not in the document]
Comment (lubosmj) [reply pulpcore/cli/ostree/distribution.py]: Would doing something like this: https://docs.pulpproject.org/pulp_cli/contributing/#version-dependent-code-paths alleviate the problem?

[the remark this answers is not in the document]
Comment (mdellweg) [reply pulpcore/cli/ostree/distribution.py]: Yes, exactly. It sounds like it is fixes in ostree, but just not yet released.

File: pulpcore/cli/ostree/distribution.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+                        body["repository_version"] = ""
+                    else:
+                        distribution_ctx.update(
+                            href, body={"repository_version": ""}, non_blocking=True
+                        )
+                body["repository"] = repository.pulp_href
+    elif version is not None:
+        # keep current repository, change version
+        if distribution["repository"]:
+            if pulp_ctx.has_plugin(PluginRequirement("ostree", min="2.0.0a2")):
```
Comment (mdellweg) [inline pulpcore/cli/ostree/distribution.py:122]: That is unreleased, no? You'd probably need to
```suggestion
            if pulp_ctx.has_plugin(PluginRequirement("ostree", min="2.0.0a2.dev")):
```

Comment (lubosmj) [reply pulpcore/cli/ostree/distribution.py:122]: 2.0.0a2 was released. I bumped the version in pulp_ostree. Then, I will change it to 2.0.0a3.dev afterwards.

Comment (mdellweg) [reply pulpcore/cli/ostree/distribution.py:122]: sounds good.

**Comment by lubosmj** [comment]
This is the first part of the CLI for pulp_ostree (it implements all basic workflows apart from the modify workflow). I am going to open the next PR right after merging this.

**Pull request merged**